python/cudf_polars/cudf_polars/utils/config.py

```diff
@@ -99,6 +100,43 @@ def __post_init__(self) -> None:
             raise TypeError("pass_read_limit must be an int")
 
 
+def default_blocksize(scheduler: str) -> int:
+    """Return the default blocksize."""
+    try:
+        # Use PyNVML to find the worker device size.
+        import pynvml
+
+        pynvml.nvmlInit()
+        index = os.environ.get("CUDA_VISIBLE_DEVICES", "0").split(",")[0]
+        if index and not index.isnumeric():  # pragma: no cover
+            # This means device_index is UUID.
+            # This works for both MIG and non-MIG device UUIDs.
+            handle = pynvml.nvmlDeviceGetHandleByUUID(str.encode(index))
+            if pynvml.nvmlDeviceIsMigDeviceHandle(handle):
+                # Additionally get parent device handle
+                # if the device itself is a MIG instance
+                handle = pynvml.nvmlDeviceGetDeviceHandleFromMigDeviceHandle(handle)
+        else:
+            handle = pynvml.nvmlDeviceGetHandleByIndex(int(index))
+
+        device_size = pynvml.nvmlDeviceGetMemoryInfo(handle).total
+
+    except (ValueError, pynvml.NVMLError):  # pragma: no cover
+        # Fall back to a conservative 8GiB default
+        device_size = 8 * 1024**3
```

Comment:
I think this is going to be difficult to debug if an NVML error is encountered here. I would suggest at a minimum adding a warning to let the user know determining the device size failed and that you're falling back to 8GiB.

Reply:
Good point - I'll add a warning here.